pkg/dnfjson/dnfjson.go

```diff
@@ -479,6 +481,20 @@ func (s *Solver) makeDepsolveRequest(pkgSets []rpmmd.PackageSet) (*Request, map[
 	return &req, rhsmMap, nil
 }
 
+func (s *Solver) optionalMetadataForDistro() []string {
+	// filelist repo metadata is required when using newer versions of libdnf
+	// with old repositories or packages that specify dependencies on files.
+	// EL10+ and Fedora 40+ packaging guidelines prohibit depending on
+	// filepaths so filelist downloads are disabled by default and are not
+	// required when depsolving for those distros. Explicitly enable the option
+	// for older distro versions in case we are using a newer libdnf.
+	switch s.modulePlatformID {
+	case "platform:f39", "platform:el8", "platform:el9":
```

Comment:
RHEL-7 should be also included, right? What about F40 worker depsolving for F38 image?

Reply:
heh, yeah, I forgot about RHEL 7 because it didn't cause any issues, most likely because the package that triggers this issue in our base images is `platform-python`.

Fedora 38 is EOL and I was planning on removing it soon.

Reply:
Ok to resolve this now?
(with an F38 removal later this week)